tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyAjax.java

```diff
@@ -1938,34 +1934,89 @@ private static Map<String, Object>[] getOldVotesRows(
+            Map<String, Object>[] rows, CLDRLocale locale, STFactory stFac, Factory diskFac) {
+        final String localeId = locale.getBaseName();
+        final Set<String> validPaths = stFac.getPathsForFile(locale);
+        final ArrayList<Map<String, Object>> al = new ArrayList<>();
+        final CLDRFile cldrFile = diskFac.make(localeId, true, true);
+        final CLDRFile cldrUnresolved = cldrFile.getUnresolved();
+        final XMLSource diskData = cldrFile.getResolvingDataSource();
+        final DisplayAndInputProcessor daip = DisplayAndInputProcessorFactory.make(locale);
         for (Map<String, Object> m : rows) {
-            int xp = (Integer) m.get("xpath");
+            /*
+             * Skip abstentions (null value) and votes for inheritance.
+             * For inheritance, a candidate item is provided anyway when appropriate.
+             */
             Object obj = m.get("value");
-            String value = (obj == null) ? null : obj.toString();
+            if (obj == null) {
+                continue;
+            }
+            String rawValue = obj.toString();
+            if (rawValue == null || rawValue.equals(CldrUtility.INHERITANCE_MARKER)) {
+                continue;
+            }
+            int xp = (Integer) m.get("xpath");
             String xpathString = CookieSession.sm.xpt.getById(xp);
-            if (!DowngradePaths.lookingAt(localeId, xpathString, value)) {
-                al.add(m);
+            if (!validPaths.contains(xpathString)) {
+                continue;
+            }
+            final String processedValue = daip.processInput(xpathString, rawValue, null);
+            if (processedValue == null) {
+                continue;
+            }
+            if (DowngradePaths.lookingAt(localeId, xpathString, processedValue)) {
+                continue;
+            }
+            final PathHeader pathHeader = stFac.getPathHeader(xpathString);
+            if (pathHeader == null || !pathHeader.canReadAndWrite()) {
+                continue;
+            }
+            if (CheckForCopy.sameAsCode(processedValue, xpathString, cldrUnresolved, cldrFile)) {
+                continue; // not allowed
             }
+            /*
+             * DAIP may have changed since old CLDR versions, and therefore what seems to be an old
+             * "losing" value may actually match the current value after DAIP in which case it should be
+             * skipped for manual import.
+             */
+            final String curValue = diskData.getValueAtDPath(xpathString);
+            if (processedValue.equals(curValue)) {
+                continue;
+            }
+            if (importTableContains(localeId, xp, processedValue)) {
+                continue;
+            }
+            al.add(m);
         }
         return al.toArray(new Map[0]);
     }
 
+    private static boolean importTableContains(String localeId, int xp, String processedValue) {
```

Comment:
This method is a last-minute addition, and makes a huge difference -- now, at least for the Filipino vetter, the manual import succeeds after "select all" and "import", with message "No old votes to import. You're done with this locale!"

Reply:
I can see this is only called after a number of conditions including that DAIP actually makes a change.  Otherwise it would be expensive to call this one by one for each value.

Reply:
@srl295 "only called after a number of conditions including that DAIP actually makes a change" -- actually, it doesn't check that. The preceding check involves the winning value (curValue), not the old losing value (rawValue).

It is expensive, so not ideal in the long run, though testing with 11,451 old votes to import it seems pretty fast.

Possibly the condition should be changed from

`if (importTableContains(localeId, xp, processedValue)) {`

to

`if (!processedValue.equals(rawValue) && importTableContains(localeId, xp, processedValue)) {`

That would be more efficient, and would have the same result -- but only if the sql query in getOldVotesRows is guaranteed to filter out any old votes whose values were already imported when `processedValue.equals(rawValue)`. Unfortunately, it's not clear to me whether that's guaranteed, since DAIP may be different for each version... Probably it would be OK but I'd rather do further study and testing rather than assume it.